### ISSUE TITLE
Add extension js to specific app template only

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#2573]
I am building larger multipage apps and the fact an extension becomes "global" as soon as I load it and included in every template is problematic because

- It takes time to load all that js 
- Some js interfer with each other.
    - For example I would like to demo both the HighChart and the HighStock extension. But their js should not be loaded in the same page. 

I thought there was already a similar issue. But I could not find it.

## Solution

Please support setting the extension for a template only like 

```python
pn.template.FastListTemplate(extensions=['vega', 'highcharts'])
```

or similar. The template should add both global extensions and the specific extensions.

## Workaround

Learning about a workaround would also be appreciated.

## Example

I'm adding a `highcharts-network` page to awesome-panel.org. And suddenly this line of code

```python
pn.extension("highchart")
```

adds the js below is in every page

![image](https://user-images.githubusercontent.com/42288570/126888171-1a67a9f1-33a0-420a-8e59-8773f248e608.png)

**Comment by philippjfr** [comment]
This would indeed be very nice but it's unfortunately also quite difficult. The way this works is that `pn.extension` simply imports the module and when Panel/Bokeh query its resources it simply iterates over all (imported) bokeh models and queries them for their dependencies. The reason this is done this way is that if you add some component to an app after initial render it still has to serve its dependencies on launch. If we now take manual control over this per app then we have to track the extensions explicitly somehow and manually importing the models may no longer work.

There's probably some solution but it's definitely not straightforward.

**Comment by MarcSkovMadsen** [comment]
Closed by #4562
